### PR TITLE
Feature/event cleanup

### PR DESCRIPTION
**Pull request by joshw**
Major event processing cleanup:

* Formalize our required event traits and provide a way to audit a ceilometer event_definitions.yaml against our requirements.
* Change event processing to work (as much as possible) with the default event_definitions.yaml, document shortcomings, and provide a minimal set of changes to be appended to event_definitions.yaml to correct these issues.
* Begin phase-out of event transforms for model change related events
* Remove all event transforms that don't update the model, and modify all mappings to drop these events rather than sending them to history (where nobody looks at them anyway)
* Change event datasource to only store events which are used to update the model, and discard any others.
* Remove all defaultmappings- these are inefficient and not necessary, since we are not going to attempt to store unknown events.  Add explicit mappings just for the event types we use to update the model.

### CODE REVIEW

File: ZenPacks/zenoss/OpenStackInfrastructure/zenpack.yaml [hunk 1]
```diff
@@ -126,7 +126,7 @@ device_classes:
           maintenance:
             type: Python
             plugin_classname: ZenPacks.zenoss.OpenStackInfrastructure.dsplugins.MaintenanceDataSourcePlugin
-            cycletime: 600
+            cycletime: 7200
```
Comment (pcarinhas) [inline ZenPacks/zenoss/OpenStackInfrastructure/zenpack.yaml]: Thats a big change from 10 mins to 2 hours.

File: ZenPacks/zenoss/OpenStackInfrastructure/events.py [hunk 2]
```diff
@@ -922,34 +922,39 @@ def event_is_mapped(evt):
...
-                for datamap in datamaps:
-                    # LOG.debug("Applying %s" % datamap)
-                    adm._applyDataMap(device, datamap)
-
-            return len(datamaps)
+        objectmap = map_event(evt)
+
+        if objectmap:
+            adm = ApplyDataMap(device)
+            # LOG.debug("Applying %s" % objectmap)
```
Comment (pcarinhas) [inline ZenPacks/zenoss/OpenStackInfrastructure/events.py]: We should either expose the LOG or remove the comment.

File: ZenPacks/zenoss/OpenStackInfrastructure/zenpack.yaml (hunk 1 above)
Comment (joshw) [reply ZenPacks/zenoss/OpenStackInfrastructure/zenpack.yaml]: Yes, deliberate.  This is a somewhat expensive operation on larger environments and it's not necessary to run it so often.

File: ZenPacks/zenoss/OpenStackInfrastructure/events.py (hunk 2 above)
Comment (joshw) [reply ZenPacks/zenoss/OpenStackInfrastructure/events.py]: Yeah, that's old code that I didn't change, but i'll fix it.

File: ZenPacks/zenoss/OpenStackInfrastructure/datasources/EventsAMQPDataSource.py [hunk 3]
```diff
@@ -101,7 +101,9 @@ def __init__(self):
 
 
 class EventsAMQPDataSourcePlugin(AMQPDataSourcePlugin):
-    proxy_attributes = ()
+    proxy_attributes = (
+        'zOpenStackProcessEventTypes'
```
Comment (pcarinhas) [inline ZenPacks/zenoss/OpenStackInfrastructure/datasources/EventsAMQPDataSource.py]: Do you need a comma here to ensure tuple?

**Pull request merged**